### PR TITLE
[ticket/12591] Improve breadcrumb naming and extend it's usage

### PR DESCRIPTION
**Pull request by senky**
PHPBB3-12591

Checklist:

- [ ] Correct branch: master for new features; 3.2.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [ ] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.2.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-12591

### CODE REVIEW

**Comment by VSEphpbb** [comment]
This will have backwards compatibility breaking consequences for any extensions with breadcrumbs. Extension like **Ideas**, **Pages**, **Board Rules** all use breadcrumbs and rely on `FORUM_NAME` and `U_VIEW_FORUM` existing in the core Prosilver templates.

Personally I don't think changing the name just for symantec reasons is worth the B.C. breakage. But if this is so desired then it should have some sort of BC fallback, i.e. set and use a template variable based on the existence of the new or the old variable names:
```
{{ set BC_NAME = BREADCRUMB_NAME || FORUM_NAME }}
{{ set BC_LINK = U_BREADCRUMB || U_VIEW_FORUM }}
```

**Comment by senky** [comment]
Funny that you advocate for original naming because I used your comment in the ticket to name variables. 😅

I did consider extensions, but this will fall into 3.3 release which can break compatibility. Your solution, however, is quite clever so I'm going for it. But we should, in any case, make a list of BC changes when releasing 3.3.0 and this should be mentioned.

File: phpBB/styles/prosilver/template/navbar_header.html [hunk 1]
```diff
@@ -186,8 +186,10 @@
 				<span class="crumb" {$MICRODATA}><a href="{U_INDEX}" itemprop="url" accesskey="h" data-navbar-reference="index"><!-- IF not U_SITE_HOME --><i class="icon fa-home fa-fw"></i><!-- ENDIF --><span itemprop="title">{L_INDEX}</span></a></span>
 
 			<!-- BEGIN navlinks -->
+				{{ set NAVINK_NAME = navlinks.BREADCRUMB_NAME || navlinks.FORUM_NAME }}
+				{{ set NAVINK_LINK = navlinks.U_BREADCRUMB || navlinks.U_VIEW_FORUM }}
```
Comment (3D-I) [inline phpBB/styles/prosilver/template/navbar_header.html]: May I ask why `NAVINK_*` isn't `NAVLINK_*`? Out of curiosity...

Comment (senky) [reply phpBB/styles/prosilver/template/navbar_header.html]: Typo. 😅

**Comment by Nicofuma** [comment]
If we can easily keep compatibility we should (and say that the older way is deprecated of course)

**Comment by senky** [comment]
It keeps compatibility.

**Pull request merged**